Assets/MRTK/SDK/Experimental/SpatialAwareness/SurfaceMeshesToPlanes.cs

```diff
@@ -0,0 +1,473 @@
+        /// <returns>Yield result.</returns>
+        private async Task MakePlanes(CancellationToken cancellationToken)
+        {
+            await new WaitForUpdate();
+
+#if PLANE_FINDING_PRESENT
+            List<PlaneFinding.MeshData> meshData = new List<PlaneFinding.MeshData>();
+            List<MeshFilter> filters = new List<MeshFilter>();
+
+            var spatialAwarenessSystem = CoreServices.SpatialAwarenessSystem;
+            if (spatialAwarenessSystem != null)
+            {
+                GameObject parentObject = spatialAwarenessSystem.SpatialAwarenessObjectParent;
+
+                // Get mesh filters from SpatialAwareness Mesh Observer
+                foreach (MeshFilter filter in parentObject.GetComponentsInChildren<MeshFilter>())
+                {
+                    filters.Add(filter);
+                }
+            }
+
+            for (int index = 0; index < filters.Count; index++)
+            {
+                MeshFilter filter = filters[index];
+                if (filter != null && filter.sharedMesh != null)
+                {
+                    // fix surface mesh normals so we can get correct plane orientation.
+                    filter.mesh.RecalculateNormals();
+                    meshData.Add(new PlaneFinding.MeshData(filter));
+                }
+            }
+
+            BoundedPlane[] planes;
+
+            await new WaitForBackgroundThread();
+            {
+                planes = PlaneFinding.FindPlanes(meshData, SnapToGravityThreshold, MinArea);
+            }
+
+            await new WaitForUpdate();
+
+            DestroyPreviousPlanes();
+            activePlanes.Clear();
+            ClassifyAndCreatePlanes(planes);
+
+            // We are done creating planes, trigger an event.
+            MakePlanesComplete?.Invoke(this, EventArgs.Empty);
+#endif // PLANE_FINDING_PRESENT
+            makingPlanes = false;
+        }
+
+#if PLANE_FINDING_PRESENT
+        /// <summary>
+        /// Create game objects to represent bounded planes in scene
+        /// </summary>
+        private void ClassifyAndCreatePlanes(BoundedPlane[] planes)
+        {
+            SetFloorAndCeilingPositions(planes);
+
+            // Create SurfacePlane objects to represent each plane found in the Spatial Mapping mesh.
+            for (int index = 0; index < planes.Length; index++)
+            {
+                BoundedPlane boundedPlane = planes[index];
+
+                // Instantiate a cube, which will have the same bounds as our BoundedPlane object.
+                GameObject destinationPlane = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                ConfigurePlaneGameObject(destinationPlane, boundedPlane);
+
+                var planeObject = new SpatialAwarenessPlanarObject();
+                planeObject.GameObject = destinationPlane;
+                planeObject.PlaneType = GetPlaneType(boundedPlane, destinationPlane);
```

Comment:
Should this section use `SpatialAwarenessPlanarObject.CreateSpatialObject` instead of the explicit constructor? (Also almost feels like a weird pattern to still have a public constructor with the `CreateSpatialObject` method, but that's not relevant here)

Reply:
There's also some overlap between `ConfigurePlaneGameObject` and `SpatialAwarenessPlanarObject.CreateSpatialObject` that'd we'd be able to dedupe and reuse existing code!

Reply:
good call

Reply:
updated